### ISSUE TITLE
mask-image: url() function doesn't work with file://

### ISSUE THREAD

**Issue by ryantownsend** [mdn/content#29106]
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/mask-image

### What specific section or headline is this issue about?

Values

### What information was incorrect, unhelpful, or incomplete?

When mentioning the `<mask-source>` can be an image loaded via `url()`, there's no mention that this doesn't work locally without a dev server (when loading HTML via `file://`), because browsers expect CORS headers to be set on the image response.

### What did you expect to see?

A warning that `url()` will not work locally via `file://`, you need to use a HTTP dev server.

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/css/mask-image`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/mask-image
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/css/mask-image/index.md
* Last commit: https://github.com/mdn/content/commit/c77cfcd17e85db6c1b93160c70668f2ff6c2809c
* Document last modified: 2023-07-20T07:38:29.000Z

</details>

**Comment by Josh-Cena** [comment]
If this is to be documented, it should probably be in https://developer.mozilla.org/en-US/docs/Web/CSS/url, unless this is specific to the place it is used in.

**Comment by ryantownsend** [comment]
@Josh-Cena it's specific to the `mask-image` attribute. [`url()`](https://developer.mozilla.org/en-US/docs/Web/CSS/url) is used by things like `background-image` which don't trigger CORS issues and therefore works with `file://` without issue.

**Comment by Josh-Cena** [comment]
This also applies to `clip-path` et al.: https://drafts.fxtf.org/css-masking/#sec We should probably document it on individual properties, then, but not sure how many there are.

**Comment by yarusome** [comment]
Related Gecko bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1565509, which is pretty informative. Basically it means that any "new" web feature should use the `cors` mode by design.

Honestly speaking, this would be quite surprising for anyone doing local testing. It is also contradictory to [2.3. Fetching External Images](https://drafts.csswg.org/css-images-4/#fetching-images) in CSS Images, where it states that images specified by `url()` are fetched with the `no-cors` mode.

See also https://github.com/whatwg/html/issues/1888 and https://github.com/w3c/csswg-drafts/issues/1481.

**Comment by ryantownsend** [comment]
@yarusome completely agree, for context, I've been in web development for 20 years and it took me by surprise.

As far as I recall, none of the browsers I was testing (Firefox, Chrome Canary, Safari) output a console error, so I was sat thinking my CSS somehow wasn't correct so I tried all manner of different combinations of attributes/values before I happened to notice the request was red in one of the network waterfalls.

Ideally, browsers should output an error in the console, but MDN coverage is the next best thing.

Thanks for supplying all this detail – it's really useful.
